dali/operators/generic/one_hot.cu

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include "dali/operators/generic/one_hot.h"
+#include "dali/operators/generic/one_hot.cuh"
+
+namespace dali {
+
+class OneHotGPU : public OneHot<GPUBackend> {
+ public:
+  explicit OneHotGPU(const OpSpec &spec) : OneHot<GPUBackend>(spec) {
+    int64_t samples_size = batch_size_ * sizeof(detail::SampleDesc);
+    scratch_mem_.set_type(TypeTable::GetTypeInfo(DALI_UINT8));
+    scratch_mem_.Resize({samples_size});
+  }
+
+  ~OneHotGPU() override = default;
+  DISABLE_COPY_MOVE_ASSIGN(OneHotGPU);
+
+  void RunImpl(workspace_t<GPUBackend> &ws) override;
+
+  template<typename OutputType, typename InputType>
+  void RunImplTyped(workspace_t<GPUBackend> &ws, int placement_axis);
+
+  USE_OPERATOR_MEMBERS();
+
+ private:
+  std::vector<detail::SampleDesc> sample_descs_;
+  Tensor<GPUBackend> scratch_mem_;
+};
+
+void OneHotGPU::RunImpl(Workspace &ws) {
+  const auto &input = ws.InputRef<GPUBackend>(0);
+  auto &output = ws.OutputRef<GPUBackend>(0);
+  int output_sample_dim = output.shape().sample_dim();
+  int placement_axis = get_placement_axis(output_sample_dim);
+  output.SetLayout(GetOutputLayout(ws, placement_axis, output_sample_dim));
+  TYPE_SWITCH(input.type().id(), type2id, InputType, ONE_HOT_TYPES, (
+    TYPE_SWITCH(output_type_, type2id, OutputType, ONE_HOT_TYPES, (
+      RunImplTyped<OutputType, InputType>(ws, placement_axis);
+    ), DALI_FAIL(make_string("Unsupported output type: ", output_type_)); );       // NOLINT
+  ), DALI_FAIL(make_string("Unsupported input type: ", input.type().id())); );     // NOLINT
+}
+
+template <typename OutputType, typename InputType>
+void OneHotGPU::RunImplTyped(workspace_t<GPUBackend> &ws, int axis) {
+  const auto &input = ws.InputRef<GPUBackend>(0);
+  auto &output = ws.OutputRef<GPUBackend>(0);
+
```

Comment:
```suggestion
auto in_view = view<const InputType>(input);
auto out_view = view<OutputType>(output);
```
and later you can just access:
`out_view.data`, `out_view.shape` and `in_view.data`, `in_view.shape` 

Just a suggestion